include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -398,9 +398,28 @@ class __future_base
     operator sycl::event() const { return __my_event; }
 };
 
-template <typename T>
+template <typename _T>
 class __future : public __future_base
 {
+    ::std::size_t __offset;
```

Comment:
I think that field should be commented.
And probably renamed, 'result_idx' for example or something like that.. it should "tell" about meaning...